crates/red_knot_python_semantic/src/types.rs

```diff
@@ -29,6 +29,76 @@ mod display;
 mod infer;
 mod narrow;
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Boundness {
+    Bound,
+    MayBeUnbound,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum SymbolLookupResult<'db> {
+    Type(Type<'db>, Boundness),
+    Unbound,
+}
```

Comment:
Nit: I typically expect that types named `Result` are aliases of the `std::result::Result` type which isn't the case. How about `SymbolType`?

Reply:
I do know what you mean, but "Result" does _feel_ like the most logical word to use here, since it's the result of attempting to lookup a symbol. Elsewhere we used the word "Outcome" for enums like this that represent the "result" of something but are not aliases of `std::result::Result`... But IIRC the previous feedback was that we didn't like that word either?

> How about `SymbolType`?

I don't like this, because this is the convention we use for structs/enums that capture data inside variants of the `Type` enum. This enum isn't that; it's very different.